src/24-hour-rule.ts

```diff
@@ -34,9 +41,45 @@ export const getMinimumOpenTime = (
   return MINIMUM_MAJOR_OPEN_TIME;
 };
 
-export const shouldPRHaveLabel = (
+/**
+ *
+ * @param {Context['github']}  github An Octokit instance
+ * @param {EventPayloads.WebhookPayloadPullRequestPullRequest} pr
+ * @returns {number} a number representing the that cation should use as the
+ * open time for the PR in milliseconds, taking draft status into account.
+ */
+export const getPROpenedTime = async (
+  github: Context['octokit'],
   pr: EventPayloads.WebhookPayloadPullRequestPullRequest,
-): boolean => {
+): Promise<number> => {
+  const [owner, repo] = pr.base.repo.full_name.split('/');
+
+  // Fetch PR timeline events.
+  const { data: events } = (await github.issues.listEventsForTimeline({
+    owner,
+    repo,
+    issue_number: pr.number,
+  })) as RestEndpointMethodTypes['issues']['listEventsForTimeline']['response'];
+
+  // Filter out all except 'Ready For Review' events.
+  const readyForReviewEvents = events
+    .sort(({ created_at: cA }, { created_at: cB }) => {
+      return new Date(cB).getTime() - new Date(cA).getTime();
+    })
+    .filter(e => e.event === 'ready_for_review');
+
+  // If this PR was a draft PR previously, set its opened time as a function
+  // of when it was most recently marked ready for review instead of when it was opened,
+  // otherwise return the PR open date.
+  return readyForReviewEvents.length > 0
+    ? new Date(readyForReviewEvents[0].created_at).getTime()
+    : new Date(pr.created_at).getTime();
+};
+
+export const shouldPRHaveLabel = async (
```

Comment:
question: do we initially tag draft PRs as ready for review when they're opened?

If they stay drafted without a `ready_for_review` event for 24 hours, will the `new-PR` label be added when it's created, removed after 24 hours, and added back when it gets un-drafted?

Reply:
@erickzhao what do you mean by tag them as ready for review? if a PR is opened as a draft, it doesn't recieve a `ready_for_review` until the first time it's explicitly marked ready for review. The new PR label has never been never added to draft PRs :)

Regular PR opened -> `new pr` label added immediately -> `new pr` label removed after requisite time elapse

Draft PR opened -> no label added -> marked ready for review -> `new pr` label added ->   `new pr` label removed after requisite time elapsed *from the point of being marked ready for review*

Reply:
Thanks for the clarification! I missed the draft-specific logic so the confusion was on my end.